roles/openshift_docker_facts/tasks/main.yml

```diff
@@ -46,10 +46,24 @@
   register: common_version
   failed_when: false
   changed_when: false
-  when: not openshift.common.is_atomic | bool
+  when: not openshift.common.is_containerized | bool
+
+- set_fact:
+    l_common_version: "{{ openshift.common.image_tag | oo_image_tag_to_rpm_version | default('0.0', True) }}"
+  when: openshift.common.is_containerized | bool
+
+- set_fact:
+    l_common_version: "{{ common_version.stdout | default('0.0', True) }}"
+  when: not openshift.common.is_containerized | bool
```

Comment:
I'd do whatever you prefer.  I sort of hate inlining if/else like this.  It almost always gets more complicated and I can no longer count how many bugs I've seen because of followup edits that subtly break things.